### ISSUE TITLE
Type of "getattr(Any, "val1", Any.val2)" is "Any | None"

### ISSUE THREAD

**Issue by Ovsyanka83** [microsoft/pyright#5068]
**Describe the bug**
Mypy reveals Any here (which makes a lot of sense) and pyright reveals Any | None.
```python
from typing import Any

def get(obj: Any) -> None:
    reveal_type(getattr(obj, "val1", obj.val2))
```

**Expected behavior**
`__getattr__` of `Any` is always `Any` and never `Optional[Any]` so the revealed type should be `Any`.

**VS Code extension or command-line**
Both vscode extension and command line.
Command line version is: 1.1.305

**Comment by erictraut** [comment]
The rules for overload resolution are not well defined by PEP 484 or any other specs. Pyright, mypy, and other Python static type checkers may [vary in small ways](https://microsoft.github.io/pyright/#/mypy-comparison?id=overload-resolution) how they interpret overloads. Pyright's handling involves some heuristics in the case of ambiguous overload selection due to arguments whose types are unknown.

That said, I think there's a reasonable argument to be made that `Unknown` would be more correct than `Any | None` in this particular case. I've adjusted pyright's heuristics accordingly. This change will be included in the next release.

**Comment by Ovsyanka83** [comment]
Big thanks! Unknown also makes sense to me.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.307, which I just published. It will also be included in a future release of pylance.
